notify_phase_end.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+
+"""
+This script is used by various workloads to notify DSI that a phase within a workload ended.
+The messages from this script are parsed by DSI's PortForwardedEventQueue in events.py.
+If profiling is enabled, this will trigger DSI to stop any profilers that are currently running.
+"""
+
+import socket
+from argparse import ArgumentParser
+import json
+
+parser = ArgumentParser()
+parser.add_argument("phase_id", type=int, help="Number of the phase that ended")
+parser.add_argument("-n", "--phase_name", type=str, help="Name of the phase that ended")
+parser.add_argument(
+    "-i", "--ip", type=str, default="0.0.0.0", help="Ip to send the notification to"
```

Comment:
I'm not going to pretend I understand why 0.0.0.0 allows this script to connect from the workload client back to the DSI host, but this is how Genny is doing and it's working :)